src/librustc_passes/rvalue_promotion.rs

```diff
@@ -159,41 +157,15 @@ impl<'a, 'gcx> CheckCrateVisitor<'a, 'gcx> {
         }
     }
 
-    fn handle_const_fn_call(&mut self, def_id: DefId,
-                            ret_ty: Ty<'gcx>, span: Span) -> Promotability {
-        if let NotPromotable = self.type_promotability(ret_ty) {
-            return NotPromotable;
-        }
-
-        let node_check = if let Some(fn_id) = self.tcx.hir.as_local_node_id(def_id) {
-            FnLikeNode::from_node(self.tcx.hir.get(fn_id)).map_or(false, |fn_like| {
-                fn_like.constness() == hir::Constness::Const
-            })
+    fn handle_const_fn_call(
+        &mut self,
+        def_id: DefId,
+    ) -> Promotability {
+        if self.tcx.is_promotable_const_fn(def_id) {
```

Comment:
This is now just a single call to `is_promotable_const_fn` which seems a major improvement over the code duplication we were doing here before.